packages/create-cloudflare/src/helpers/args.ts

```diff
@@ -1,88 +1,239 @@
+};
+
+export type ArgumentsDefinition = {
+	intro: string;
+	positionals?: ArgDefinition[];
+	options?: OptionDefinition[];
+};
+
+const cliDefinition: ArgumentsDefinition = {
+	intro: `
+    The create-cloudflare cli (also known as C3) is a command-line tool designed to help you set up and deploy new applications to Cloudflare. In addition to speed, it leverages officially developed templates for Workers and framework-specific setup guides to ensure each new application that you set up follows Cloudflare and any third-party best practices for deployment on the Cloudflare network.
+  `,
+	positionals: [
+		{
+			name: "directory",
 			type: "string",
-			description:
-				"The directory where the application should be created. The name of the application is taken from the directory name",
-		})
-		.option("type", {
+			description: `The directory where the application should be created. Also used as the name of the application.
+
+        If a path is provided that includes intermediary directories, only the base name will be used as the name of the application.`,
+		},
+	],
+	options: [
+		{
+			name: "type",
+			alias: "t",
 			type: "string",
-			requiresArg: true,
-			description: "The type of application that should be created",
-		})
-		.option("framework", {
+			description: `
+        When using a built-in template, specifies the type of application that should be created.
+
+        Note that "--type" and "--template" are mutually exclusive options. If both are provided, "--type" will be used.
+        `,
+			values: [
+				{
+					name: "web-framework",
+					description: "A website or web application.",
+				},
+				{
+					name: "hello-world",
+					description: "A basic “Hello World” Cloudflare Worker.",
+				},
+				{
+					name: "hello-world-durable-object",
+					description:
+						"A basic “Hello World” Cloudflare Worker with a Durable Worker.",
+				},
+				{
+					name: "common",
+					description:
+						"A Cloudflare Worker which implements a common example of routing/proxying functionalities.",
+				},
+				{
+					name: "scheduled",
+					description:
+						"A scheduled Cloudflare Worker (triggered via Cron Triggers).",
+				},
+				{
+					name: "queues",
+					description:
+						"A Cloudflare Worker which is both a consumer and produced of Queues.",
+				},
+				{
+					name: "openapi",
+					description: "A Worker implementing an OpenAPI REST endpoint.",
+				},
+			],
+		},
+		{
+			name: "framework",
+			alias: "f",
 			type: "string",
-			requiresArg: true,
-			description:
-				"The type of framework to use to create a web application (when using this option `--type` is ignored)",
-		})
-		.option("deploy", {
+			description: `The type of framework to use to create a web application (when using this option "--type" is coerced to "web-framework")
+
+      When using the --framework option, C3 will dispatch to the official creation tool used by the framework (ex. "create-remix" is used for Remix).
```

Comment:
it feels almost like we give Remix a preferential treatment here 😅

Reply:
Yeah, but having an in-context usage example seems useful. Any thoughts on an alternative?